### PR TITLE
Update heasarc image to heasoft v6.30.1

### PR DESCRIPTION
**Pull request by zoghbi-a**
This PR updates the heasarc image to use heasoft v6.30. This is the first version where we have a version of heasoftpy with external python dependencies, so heasoft now has its own conda environment. The image uses a fresh download of miniconda to speed up the build process. The tests test/ work when run manually. Here is a list of the changes:

- Start with a fresh miniconda with python=3.8
- heasoft is installed in a conda environment called heasoft. 
- XMM-SAS has its own conda environment too to handle its own python dependencies.
- There are now 4 conda environments: heasoft, ciao, fermi and xmmsas.
- Added scripts to `CONDA/env/{env}/etc/conda/activate.d/` that initialize heasoft and xmmsas when their respective conda environments are activated. ciao and fermi have their own scripts to do that automatically.
- Jupyter is run from the heasoft environment, which is the default environment.
- HEADAS points to `/opt/heasoft/x86_64-pc-linux-gnu-libc2.27 not /opt/heasoft`. The latter breaks `pyxspec`. XMM-SAS is also installed to `/opt/xmmsas/xmmsas_20211130_0941` not `/opt/xmmsas`

### CODE REVIEW

File: heasarc/test/test_all.sh [hunk 1]
```diff
@@ -164,10 +164,13 @@ echo ""
 echo "#############  XMM SAS stuff #########"
 echo "              (under development) "
 
+echo "  (switching to xmmsas environment) "
+conda activate xmmsas
 
 echo "###  running SAS setup"
 export SAS_CCFPATH=/FTP/caldb/data/xmm/ccf
-source /opt/xmmsas/setsas.sh
+source /opt/xmmsas/xmmsas_*/setsas.sh
```
Comment (trjaffe) [inline heasarc/test/test_all.sh:172]: Shouldn't be needed anymore since it's run when the env is ativated, right?

File: heasarc/test/test_all.sh [hunk 2]
```diff
@@ -7,9 +7,9 @@
 set -e
 
 xte=true
-ciao=true
-fermi=true
-xmm=true
+ciao=false
+fermi=false
+xmm=false
```
Comment (trjaffe) [inline heasarc/test/test_all.sh:12]: I never got these working fully automatically, but it looks like you might have?

**Comment by trjaffe** [comment]
For me, this is failing at the step where it installs the wwtelescope app.  No idea why.  I'm commenting it out so I can test the rest.

**Comment by trjaffe** [comment]
I haven't had time.  I think I need to go on center to be able to reconfigure my Docker desktop.  I think Manu might as well build it and then we can test it there.

File: heasarc/sciserver_heasoft/Dockerfile [hunk 3]
```diff
@@ -84,9 +84,18 @@ RUN cd /home/${sciserver_user}/heasoft-${HEASOFT_VERSION}/BUILD_DIR/ \
  && make install 2>&1 | tee /home/${sciserver_user}/install.log
 ## ------------------------------- ##
 
+## ------------------ ##
+## Apply patch 6.30.1 
+RUN cd /home/${sciserver_user} \
+ && wget https://heasarc.gsfc.nasa.gov/FTP/software/lheasoft/lheasoft6.30/updates/hdpatch_6.30.1 \
```
Comment (trjaffe) [inline heasarc/sciserver_heasoft/Dockerfile:90]: Bryan found an issue with the original patch.  Did he replace it at the same version number?

Comment (esabol) [reply heasarc/sciserver_heasoft/Dockerfile:90]: Why are you even using the patch? Just download and install the 6.30.1 tarball by specifying the HEASOFT_VERSION to be 6.30.1.

Comment (zoghbi-a) [reply heasarc/sciserver_heasoft/Dockerfile:90]: @esabol. I am doing that. I initially thought it cannot be done. Thanks for the note.

**Comment by zoghbi-a** [comment]
This last commit updates heasoft to v6.30.1.
I have also made changes to reduce the size of the final images. 
For heasoft, ftools/xstar/data/atdb.fits is ~1GB, so it is removed and replaced by a soft link to the data store in FTP/.
For ciao, CALDB is also linked to FTP/caldb instead of downloading it from the ciao conda channel.

**Pull request merged**